src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -318,6 +321,209 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
+        var randomIntPattern = new Regex(@$"\{"$randomInt"}(?:\s(?<parameters>-?\d+)){{0,2}}", RegexOptions.Compiled);
+        var timestampPattern = new Regex(@$"\{"$timestamp"}(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+
+        var guidMatches = guidPattern.Matches(expression);
+        var dateTimeMatches = dateTimePattern.Matches(expression);
+        var localDateTimeMatches = localDateTimePattern.Matches(expression);
+        var randomIntMatches = randomIntPattern.Matches(expression);
+        var timestampMatches = timestampPattern.Matches(expression);
+
+        if (guidMatches.Count > 0)
+        {
+            return node.CreateBindingSuccess(Guid.NewGuid().ToString());
+        } else if(dateTimeMatches.Count > 0)
+        {
+            return GetDateTime(node, expression, dateTimeMatches, dateTimePattern);
+        } else if(localDateTimeMatches.Count > 0)
+        {
+            return GetDateTime(node, expression, localDateTimeMatches, localDateTimePattern);
+        } else if(randomIntMatches.Count > 0)
+        {
+            return GetRandInt(node, expression, randomIntMatches);
+        }else if(timestampMatches.Count > 0)
+        {
+            return GetTimestamp(node, expression, timestampMatches);
+        }
+        return null;
+    }
+
+    private HttpBindingResult<object?> GetTimestamp(HttpExpressionNode node, string expression, MatchCollection matches)
+    {
+        var text = expression;
+        var currentDateTimeOffset = DateTimeOffset.UtcNow;
+        
+        DateTimeOffset dateTimeOffset = currentDateTimeOffset;
+        var match = matches.FirstOrDefault();
+        if (match?.Groups.Count == 3)
+        {
+            try
+            {
+                if (match.Groups["offset"].Success && match.Groups["option"].Success
+                    && int.TryParse(match.Groups["offset"].Value, out int offset)
+                    && offset != 0)
+                {
+                    dateTimeOffset = dateTimeOffset.AddOffset(offset, match.Groups["option"].Value);
+                }
+
+                text = string.Concat(text.Substring(0, match.Index), dateTimeOffset.ToUnixTimeSeconds().ToString(), text.Substring(match.Index + match.Value.Length));
+            }
+            catch (ArgumentException)
+            {
+                return node.CreateBindingFailure(HttpDiagnostics.InvalidOffset(match.Groups["offset"].Value));
+            }
+            return node.CreateBindingSuccess(text);
+        } else
+        {
+            return node.CreateBindingFailure(HttpDiagnostics.TimestampFormatError(expression));
+        }
+        
+    }
+
+    private HttpBindingResult<object?> GetDateTime(HttpExpressionNode node, string expression, MatchCollection matches, Regex pattern)
+    {
+        var text = expression;
```

Comment:
nit: Avoid the new var is redundant. I would recommend just using the parameter `expression` - you could rename the parameter to `expressionText` or `dateTimeExpressionText` if you want for clarity.